engine/runtime/src/main/scala/org/enso/compiler/pass/resolve/TypeNames.scala

```diff
@@ -44,7 +44,19 @@ case object TypeNames extends IRPass {
       ir.unsafeGetMetadata(BindingAnalysis, "bindings analysis did not run")
     ir.copy(bindings = ir.bindings.map { d =>
       val mapped = d.mapExpressions(resolveExpression(bindingsMap, _))
-      doResolveType(bindingsMap, mapped)
+      doResolveType(
+        bindingsMap,
+        mapped match {
+          case typ: IR.Module.Scope.Definition.Type =>
+            typ.members.foreach(m => {
+              m.arguments.foreach(a => {
+                doResolveType(bindingsMap, a)
+              })
+            })
+            typ
+          case x => x
+        }
```

Comment:
```suggestion
        mapped match {
          case typ: IR.Module.Scope.Definition.Type =>
            typ.members.foreach(_.arguments.foreach(a =>
                doResolveType(bindingsMap, a)
              )
            )
            typ
          case x => x
        }
```

Reply:
Done in [17d7453](https://github.com/enso-org/enso/pull/6584/commits/17d745393a204efb06b507e72036eda8fc1c0e30)